### PR TITLE
Fix filters with complex object are not removed from the UI

### PR DESCRIPTION
**Pull request by hmatthieu**
When filters with a complex object as defaultValue the "empty value" was not considered empty by the UI. This commit ensure we don't display empty filters on the UI.

Refs: #9893

### CODE REVIEW

**Comment by hmatthieu** [comment]
I reviewed the logic around the reset form behavior a bit. In my opinion, it could be simplified, but I don't want to risk removing code that seems overkill. The changes I made in this PR aim to reduce breaking changes and simply address the UI issue that was originally reported.

**Comment by djhi** [comment]
Thanks for the PR! Would you mind adding a test for it?

**Comment by hmatthieu** [comment]
Tests added

**Comment by adguernier** [comment]
Thanks for your work :muscle: Waiting for another review

**Review by djhi — APPROVED** [review]

**Comment by hmatthieu** [comment]
I'll review the failing test

**Comment by hmatthieu** [comment]
I added a fix (to fix the test :x). I tested on my local setup, with a real chrome & firefox (managed by cypress). Was working properly. From my XP, those failing tests comes from CPU perfs and bad starting conditions for the test. I added a condition where we wait for an element to be present (which is safe than waiting for an element that is not present).

I also removed a `.only` that was forgotten there.

**Review by adguernier — APPROVED** [review]

File: cypress/e2e/mobile.cy.js
Comment (djhi) [inline cypress/e2e/mobile.cy.js:12]: :scream:

**Pull request merged**
